From 08bb423f3b5332de14fe41e5c61340aa00fefe16 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Wed, 12 Nov 2014 16:06:34 -0500
Subject: [PATCH 716/933] Fix coverity warning about sizeof(void **) !=
 sizeof(void *)

The size of the copied data is actually void *, not void **.

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 drivers/staging/lttng/probes/lttng-events.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/staging/lttng/probes/lttng-events.h b/drivers/staging/lttng/probes/lttng-events.h
index ce4c3f1..0c92c17 100644
--- a/drivers/staging/lttng/probes/lttng-events.h
+++ b/drivers/staging/lttng/probes/lttng-events.h
@@ -438,8 +438,8 @@ static inline size_t __event_get_size__##_name(size_t *__dynamic_len,	      \
 		const void *__ctf_tmp_ptr = (_src);			       \
 		memcpy(__stack_data, &__ctf_tmp_ulong, sizeof(unsigned long)); \
 		__stack_data += sizeof(unsigned long);			       \
-		memcpy(__stack_data, &__ctf_tmp_ptr, sizeof(void **));	       \
-		__stack_data += sizeof(void **);			       \
+		memcpy(__stack_data, &__ctf_tmp_ptr, sizeof(void *));	       \
+		__stack_data += sizeof(void *);				       \
 	}
 
 #undef _ctf_sequence_encoded
@@ -450,16 +450,16 @@ static inline size_t __event_get_size__##_name(size_t *__dynamic_len,	      \
 		const void *__ctf_tmp_ptr = (_src);			       \
 		memcpy(__stack_data, &__ctf_tmp_ulong, sizeof(unsigned long)); \
 		__stack_data += sizeof(unsigned long);			       \
-		memcpy(__stack_data, &__ctf_tmp_ptr, sizeof(void **));	       \
-		__stack_data += sizeof(void **);			       \
+		memcpy(__stack_data, &__ctf_tmp_ptr, sizeof(void *));	       \
+		__stack_data += sizeof(void *);				       \
 	}
 
 #undef _ctf_string
 #define _ctf_string(_item, _src, _user, _nowrite)			       \
 	{								       \
 		const void *__ctf_tmp_ptr = (_src);			       \
-		memcpy(__stack_data, &__ctf_tmp_ptr, sizeof(void **));	       \
-		__stack_data += sizeof(void **);			       \
+		memcpy(__stack_data, &__ctf_tmp_ptr, sizeof(void *));	       \
+		__stack_data += sizeof(void *);				       \
 	}
 
 #undef TP_PROTO
-- 
2.0.2

